sensor/common/image/service_impl_test.go

```diff
@@ -0,0 +1,239 @@
+			notify:           common.SensorComponentEventOfflineMode,
+			expectCache:      expectCacheHelper(s.mockCache, 1, nil),
+			expectedError:    errCentralNoReachable,
+			expectedResponse: nil,
+		},
+		"Cache miss, local scan, central is reachable": {
+			request:          createImageRequest(imageName, imageID, false),
+			notify:           common.SensorComponentEventCentralReachable,
+			expectCache:      expectCacheHelper(s.mockCache, 1, nil),
+			expectRegistry:   expectRegistryHelper(s.mockRegistryStore, 1, true),
+			expectLocalScan:  expectLocalScan(s.mockLocalScan, 1, createScannedImage(imageName, imageID), nil),
+			expectedError:    nil,
+			expectedResponse: createImageResponse(imageName, imageID),
+		},
+		"Cache miss, local scan returns error, central is reachable": {
+			request:          createImageRequest(imageName, imageID, false),
+			notify:           common.SensorComponentEventCentralReachable,
+			expectCache:      expectCacheHelper(s.mockCache, 1, nil),
+			expectRegistry:   expectRegistryHelper(s.mockRegistryStore, 1, true),
+			expectLocalScan:  expectLocalScan(s.mockLocalScan, 1, nil, err),
+			expectedError:    errLocalScan,
+			expectedResponse: nil,
+		},
+	}
+	for testName, c := range cases {
+		s.Run(testName, func() {
+			s.createImageService()
+			s.service.Notify(c.notify)
+			c.expectCache.run()
+			c.expectRegistry.run()
+			c.expectCentralCall.run()
+			c.expectLocalScan.run()
+			res, err := s.service.GetImage(ctx, c.request)
+			if c.expectedError != nil {
+				s.Assert().EqualError(err, c.expectedError.Error())
+			} else {
+				s.Assert().NoError(err)
+			}
+			s.Assert().Equal(c.expectedResponse, res)
+		})
+	}
+}
+
+func expectCacheHelper(mockCache *cacheMocks.MockCache, times int, retValue any) expectFn {
+	return func() {
+		mockCache.EXPECT().Get(gomock.Any()).Times(times).DoAndReturn(func(_ any) any {
+			return retValue
+		})
+	}
+}
+
+func expectRegistryHelper(mockRegistryStore *imageMocks.MockregistryStore, times int, retValue bool) expectFn {
+	return func() {
+		mockRegistryStore.EXPECT().IsLocal(gomock.Any()).Times(times).DoAndReturn(func(_ any) bool {
+			return retValue
+		})
+	}
+}
+
+func expectCentralCall(mockCentral *imageMocks.MockcentralClient, times int, retValue *v1.ScanImageInternalResponse, retErr error) expectFn {
+	return func() {
+		mockCentral.EXPECT().ScanImageInternal(gomock.Any(), gomock.Any()).Times(times).DoAndReturn(func(_, _ any, _ ...any) (*v1.ScanImageInternalResponse, error) {
+			return retValue, retErr
+		})
+	}
+}
+
+func expectLocalScan(mockLocalScan *imageMocks.MocklocalScan, times int, retValue *storage.Image, retErr error) expectFn {
+	return func() {
+		mockLocalScan.EXPECT().EnrichLocalImageInNamespace(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Times(times).DoAndReturn(func(_, _, _, _, _, _ any) (*storage.Image, error) {
+			return retValue, retErr
+		})
+	}
+}
+
+type expectFn func()
+
+func (f expectFn) run() {
```

Comment:
WDYT about `runIfNotNil` or `maybeRun` or `tryAndRun` or `runIfSet`

Reply:
I've renamed it to `runIfSet`